cucumber/authenticators/features/authn_status.feature

```diff
@@ -0,0 +1,257 @@
+Feature: Authenticator status check
+
+  Scenario: A healthy authenticator returns a successful response
+    Given a policy:
+    """
+    - !policy
+      id: conjur/authn-oidc/keycloak
+      body:
+      - !webservice
+        annotations:
+          description: Authentication service for Keycloak, based on Open ID Connect.
+
+      - !webservice
+        id: status
+        annotations:
+          description: Status service to verify the authenticator is configured correctly
+
+      - !variable
+        id: provider-uri
+
+      - !variable
+        id: id-token-user-property
+
+      - !group users
+
+      - !permit
+        role: !group users
+        privilege: [ read, authenticate ]
+        resource: !webservice
+
+      - !group
+        id: managers
+        annotations:
+          description: Group of users who can check the status of the authn-oidc/keycloak authenticator
+
+      - !permit
+        role: !group managers
+        privilege: [ read ]
+        resource: !webservice status
+
+    - !user alice
+
+    - !grant
+      role: !group conjur/authn-oidc/keycloak/users
+      member: !user alice
+
+    - !grant
+      role: !group conjur/authn-oidc/keycloak/managers
+      member: !user alice
+    """
+    And I am the super-user
+    And I successfully set OIDC variables
+    And I login as "alice"
+    When I GET "/authn-oidc/keycloak/cucumber/status"
+    Then the HTTP response status code is 200
+    And the authenticator status check succeeds
+
+  Scenario: An unauthorized user is responded with 403
+    Given a policy:
+    """
+    - !policy
+      id: conjur/authn-oidc/keycloak
+      body:
+      - !webservice
+        annotations:
+          description: Authentication service for Keycloak, based on Open ID Connect.
+
+      - !webservice
+        id: status
+        annotations:
+          description: Status service to verify the authenticator is configured correctly
+
+      - !variable
+        id: provider-uri
+
+      - !variable
+        id: id-token-user-property
+
+      - !group users
+
+      - !permit
+        role: !group users
+        privilege: [ read, authenticate ]
+        resource: !webservice
+
+      - !group
+        id: managers
+        annotations:
+          description: Group of users who can check the status of the authn-oidc/keycloak authenticator
+
+      - !permit
+        role: !group managers
+        privilege: [ read ]
+        resource: !webservice status
+
+    - !user alice
+
+    - !grant
+      role: !group conjur/authn-oidc/keycloak/users
+      member: !user alice
+    """
+    And I am the super-user
+    And I successfully set OIDC variables
+    And I login as "alice"
+    When I GET "/authn-oidc/keycloak/cucumber/status"
+    Then the HTTP response status code is 403
+    And the authenticator status check fails with error "#<Errors::Authentication::Security::UserNotAuthorizedInConjur: CONJ00006E User 'alice' is not authorized in the Conjur policy>"
+
```

Comment:
One of the advantages of having custom errors for everything is that you don't have to assert on specific error strings.  Doing so makes tests brittle.  Changing the specific text of an error shouldn't break a tests.

In this case should be enough to write:

```
And the authenticator status check fails with error UserNotAuthorizedInConjur
```
and have the step def merely check that `UserNotAuthorizedInConjur` is a substring of the error message.
```

Reply:
Agree, thanks

Reply:
Were you still planning to change this one?